res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+        inKey: "volume",
+    });
+    this.orientationLeftBtn = new components.Button({
+        midi: [0x93, 0x04 + deckNumber],
+        group: "[Channel" + deckNumber +"]",
+        key: "orientation",
+        input: function(_channel, _control, value, _status, _group) {
+            if (!value) { return; }
+            this.inSetValue(this.inGetValue() !== 0 ? 0 : 1);
+        },
+        output: function(value, _group, _control) {
+            this.send(value === 0 ? this.on : this.off);
+        },
+    });
+    this.orientationRightBtn = new components.Button({
+        midi: [0x93, 0x08 + deckNumber],
+        group: "[Channel" + deckNumber +"]",
+        key: "orientation",
+        input: function(_channel, _control, value, _status, _group) {
+            if (!value) { return; }
+            this.inSetValue(this.inGetValue() !== 2 ? 2 : 1);
+        },
+        output: function(value, _group, _control) {
+            this.send(value === 2 ? this.on : this.off);
+        },
+    });
+    this.reconnectComponents();
+};
+g4v.MixerStrip.prototype = new components.ComponentContainer();
+
+g4v.OtherControls = function() {
+    components.ComponentContainer.call();
+    this.libraryEnc = new components.Encoder({
+        midi: [0xB3, 0x1E],
+        group: "[Library]",
+        shift: function() { this.inKey = "MoveHorizontal"; },
+        unshift: function() { this.inKey = "MoveVertical"; },
+        inValueScale: function(value) { return value === 0x41 ? 1 : -1; },
+        input: function(_channel, _control, value, _status, _group) {
+            if (engine.getValue("[PreviewDeck1]", "play")) {
+                engine.setValue("[PreviewDeck1]", (value === 0x41 ? "beatjump_4_forward" : "beatjump_4_backward"), 1);
+            } else {
+                this.inSetParameter(this.inValueScale(value));
+            }
+        },
+    });
+    this.libraryBtn = new components.Component({
+        group: "[Library]",
+        shift: function() {
+            this.input = function(_channel, _control, value, _status, _group) {
+                if (value === 0x00) { return; }
+                if (engine.getValue("[PreviewDeck1]", "play", 1)) {
+                    engine.setValue("[PreviewDeck1]", "play", 0);
+                    engine.beginTimer(
+                        100, function() {
+                            engine.setValue("[PreviewDeck1]", "eject", 1);
+                            engine.setValue("[PreviewDeck1]", "eject", 0);
+                        },
+                        true);
+                } else {
+                    engine.setValue("[PreviewDeck1]", "LoadSelectedTrack", 1);
+                    engine.setValue("[PreviewDeck1]", "play", 1);
+                    engine.beginTimer(100, function() { engine.setValue("[PreviewDeck1]", "play", 1); }, true);
+                }
+            };
+        },
+        unshift: function() {
+            this.input = components.Component.input;
```

Comment:
```suggestion
            this.input = components.Component.prototype.input;
```
Not sure if it even works in the current state but this is more robust.

Reply:
Also, this component has no key defined for the unshifted output so this also doesn't work currently.